pkg/database/database.go

```diff
@@ -52,7 +53,18 @@ type Database struct {
 	Deleter  facade.DBDeleter
 }
 
+var db DBInterface
```

Comment:
Should this be a pointer?

Reply:
yes I think it's a good idea to also tackle this change in the PR

Reply:
I'm not suggesting any radical change here. Since this is a singleton the thing getting passed around shouldn't be copied. I don't know what happens if you pass an interface around, does that automatically pass the pointer of the underlying object? Or we need to make this explictly a pointer. Also, I'm not sure if we can make a pointer out of a interface.

Reply:
ah, right! I answered too quickly! The interfaces shouldn't be used as pointers (generally speaking) as they are already pointers to what's implemented so, using a pointer in here it's like returning a pointer to a pointer.

You can check how interfaces work [here](https://research.swtch.com/interfaces). Note specifically this sentence:

> Interface values are represented as a two-word pair giving a pointer to information about the type stored in the interface and a pointer to the associated data. Assigning b to an interface value of type Stringer sets both words of the interface value.

Reply:
Then yes, it should be kept as it.